experimental/InvariantTheory/AlgebraicGroups.jl

```diff
@@ -0,0 +1,218 @@
+    as a subscheme of 𝔸ⁿ(𝕜);
+  * a t×t-matrix `Z` of type `MatrixType` with entries in `R` embedding 
+    G as an algebraic subscheme of GL(t,𝕜) and endowing it with its group structure.
+"""
+mutable struct AlgGroup{FieldType, RingType, RingElemType, MatrixType}
+  k::FieldType
+  R::RingType 			# The underlying polynomial ring
+  I::MPolyIdeal{RingElemType}	# The ideal describing the closed subgroup
+  Z::MatrixType			# The matrix coordinates for the representation
+
+  function AlgGroup( k::FieldType, R::RingType, I::MPolyIdeal{RingElemType}, Z::MatrixType) where{FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+    @assert typeof(Z) <: dense_matrix_type(R) "Matrix is not defined over the given ring"
+    @assert base_ring(R) === k "ambient polynomial ring is not defined over the given field" 
+    @assert MatrixType == dense_matrix_type(R) "Matrix has the wrong type"
+    return new{typeof(k), typeof(R), elem_type(R), dense_matrix_type(R)}(k, R, I, Z)
+  end
+end
+
+AlgGroup(R::RingType, I::MPolyIdeal{RingElemType}, Z) where{RingType<:Ring, RingElemType} = AlgGroup(base_ring(R), R, I, Z)
+
+function AlgGroup(I::MPolyIdeal{RingElemType}, Z) where{RingElemType}
+  @assert typeof(Z) == dense_matrix_type(base_ring(I)) 
+  @assert base_ring(Z) == base_ring(Z) "matrix is not defined over the same ring as the given ideal"
+  return AlgGroup(base_ring(base_ring(Z)), base_ring(Z), I, Z)
+end
+
+@Markdown.doc """
+    base_field(G::AlgGroup)
+
+Returns the field 𝕜 over which G is defined.
+"""
+function base_field(G::AlgGroup{FieldType, RingType, RingElemType, MatrixType}) where{
+	FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+  return G.k::FieldType
+end
+
+@Markdown.doc """
+    ambient_ring(G::AlgGroup)
+
+Returns the polynomial ring `R` used to define `G` as an affine variety.
+"""
+function ambient_ring(G::AlgGroup{FieldType, RingType, RingElemType, MatrixType}) where{
+	FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+  return G.R::RingType
+end
+
+@Markdown.doc """
+    defining_ideal(G::AlgGroup)
+
+Returns the ideal `I` used to define `G` as an affine subvariety of 𝔸ⁿ(𝕜).
+"""
+function defining_ideal(G::AlgGroup{FieldType, RingType, RingElemType, MatrixType}) where{
+	FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+  return G.I::MPolyIdeal{elem_type(G.R)}
+end
+
+@Markdown.doc """
+    presentation_matrix(G::AlgGroup)
+
+Returns the t×t-matrix `Z` with entries in `R` for the variety G = Spec `R`/`I` 
+which presents G as a subgroup of GL(t,𝕜).
+"""
+function presentation_matrix(G::AlgGroup{FieldType, RingType, RingElemType, MatrixType}) where{
+	FieldType<:Field, RingType<:Ring, RingElemType, MatrixType}
+  return G.Z::dense_matrix_type(G.R)
+end
+
+@Markdown.doc """
+    GL( n::Int, k::FieldType ) where{FieldType<:Field}
+
+The general linear group of invertible n×n-matrices over the field `k`.
+"""
+function AlgGroup_GL( n::Int, k::FieldType ) where{FieldType<:Field}
```

Comment:
```suggestion
function AlgGroup_GL( n::Int, k::Field )
```
Of course the same should be done in various other places